lib/domain/transients/update_collection_metadata.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:memo/application/widgets/theme/rich_text_field.dart';
+
+/// Holds `Collection` properties when updating a `Collection`.
+@immutable
+class CollectionUpdateMetadata extends Equatable {
+  const CollectionUpdateMetadata({required this.name, required this.description, required this.tags});
+
+  factory CollectionUpdateMetadata.empty() =>
+      const CollectionUpdateMetadata(name: '', description: MemoUpdateContent(), tags: []);
+
+  final String name;
+  final MemoUpdateContent description;
+  final List<String> tags;
+
+  CollectionUpdateMetadata copyWith({String? name, MemoUpdateContent? description, List<String>? tags}) =>
+      CollectionUpdateMetadata(
+        name: name ?? this.name,
+        description: description ?? this.description,
+        tags: tags ?? this.tags,
+      );
+
+  @override
+  List<Object?> get props => [name, tags, description];
+}
+
+/// Holds `Memo` properties when updating a `Collection`.
+@immutable
+class MemoUpdateMetadata extends Equatable {
+  const MemoUpdateMetadata({required this.question, required this.answer});
+
+  factory MemoUpdateMetadata.empty() =>
+      const MemoUpdateMetadata(question: MemoUpdateContent(), answer: MemoUpdateContent());
+
+  MemoUpdateMetadata copyWith({MemoUpdateContent? question, MemoUpdateContent? answer}) =>
+      MemoUpdateMetadata(question: question ?? this.question, answer: answer ?? this.answer);
+
+  final MemoUpdateContent question;
+  final MemoUpdateContent answer;
+
+  @override
+  List<Object?> get props => [question, answer];
+}
+
+/// Holds `Memo` question/answer properties when updating a `Collection`.
+@immutable
+class MemoUpdateContent extends Equatable {
```

Comment:
@matuella Created this transient `MemoUpdateContent` and moved the others MemoUpdateCollection` and `MemoUpdateMetadata` to here as well

Reply:
Does it make sense to move them to domain/transient, if they only belong to that particular UI's view model?

Reply:
That's related to the questions that we've about what a transient model is. This entities are reused between different UI/View Models, so it deserves a dedicated file so all common files can import it (otherwise we would have a VM importing another VM just to get this entities).

We have defined as transients:

> Just like the models/, the transients represent a set of business structures, although there's a clear distinction between these and models - they exist only in memory, meaning that they make sense only in some particular contexts required by the application's lifecycle.

Following this definition, makes sense to this models be here IMO, what do you think?

Reply:
My only concern with this is that while "multiple" VMs use the same structure, it still belongs to this "update" flow, which could be considered a "single thing" in terms of application-wide structures.

The definition of "transients" are pretty weak right now and I think, like a couple of other architectural decisions, that we should revisit them, though it's tangential to this PR. Given this issue, I would suggest that we create a dedicated file for this structure in the same folder of this application/ domain category, which would be home/collections/update

Reply:
suggestion though

Reply:
Agreed